File: teamberg_omimi.py

def task_output(name, email, slack_username, biostack, twitter_handle, hamming_distance):
  print("name: "+ name + "\n" +"email: "+ email + "\n" +"slack_username: "+ slack_username+ "\n"+"biosatck: "+ biostack + "\n" + "twitter_handle: "+ twitter_handle + "\n" + "hamming_distance: "+ hamming_distance )


def hamming_distance(slack_username, twitter_username):
    counter = 0
    for x in range(len(slack_username)):
        if slack_username[x] != twitter_username[x]:
            counter+=1
    return str(counter)




if __name__ == '__main__':
    task_output("barnabas oretan","barnabas.oretan@gmail.com","@omimi","genomics","@Oshola_omimi",
    hamming_distance("@omimi", "@Oshola_omimi"))

File: teamberg_Omimi.cpp
#include <iostream>
#include <string>
#include <stdexcept>

using namespace std;

int hammingDistance(string &, string &);
int main(){
    string name = "Barnabas Oretan";
    string email = "barnabas.oretan@gamil.com";
    string biostack= "genomics";
    string slackUsername= "@omimiIII";
    string twitterHandle= "@omimi_II";
    try{
        int distance = hammingDistance(slackUsername, twitterHandle);
        cout << name<<","<< email << "," << slackUsername<< "," << biostack<< "," << twitterHandle<< "," << distance << endl;
    }catch (invalid_argument &invalid_argument){
        cout << invalid_argument.what()<< endl;
    }
    return 0;
    
} // namespace std;
int hammingDistance(string & slackUsername, string & twitterHandle){
    int distanceCounter = 0;
    if (slackUsername.length()!= slackUsername.length()){
        throw invalid_argument("your inputs must be of the same length");
    }
    for (size_t i = 0; i < slackUsername.length(); ++i){
        if (slackUsername[i] != twitterHandle[i]){
            ++distanceCounter;
        }
    }
    return distanceCounter;
}
